src/vicsetup/libvicsetup/defs.h

```diff
@@ -0,0 +1,37 @@
+#ifndef _VIC_DEFS_H
+#define _VIC_DEFS_H
+
+#define VIC_INLINE static __inline__
+
+#define VIC_PACK_BEGIN _Pragma("pack(push, 1)")
+#define VIC_PACK_END _Pragma("pack(pop)")
+
+#define VIC_COUNTOF(ARR) (sizeof(ARR) / sizeof(ARR[0]))
+
+#define VIC_OFFSETOF(TYPE, MEMBER) __builtin_offsetof(TYPE, MEMBER)
+
+#define VIC_UNUSED __attribute__((unused))
+
+#define __LUKS_CONCAT(X, Y) X##Y
+#define VIC_CONCAT(X, Y) __LUKS_CONCAT(X, Y)
+
+#define VIC_STATIC_ASSERT(COND) \
+    typedef char VIC_CONCAT(__vic_static_assert, __LINE__)[(COND) ? 1 : -1] \
```

Comment:
Same comment on double underscores.  For any vaguely modern compiler, this should be defined at `_Static_assert(COND)`.  See this complete implementation:

https://github.com/freebsd/freebsd/blob/6af4a8cb884975f65a51b3b6d766e084faf55d4f/sys/sys/cdefs.h#L285

(Note that this is part of libc and so is permitted to use double-underscore identifiers)